### DISCUSSION TITLE
Monit support to monitor and restart zigbee2mqtt service

### DISCUSSION THREAD

**Discussion by lansi002** [Koenkk/zigbee2mqtt#9125] [General]
## Is your feature request related to a problem? Please describe

I would like to monitor zigbee2mqtt with monitoring tool Monit.

I use the following configuration in monit.rc for monitoring Domoticz.
    _check process domoticz-service with pidfile /var/run/domoticz.pid
   start program = "/home/pi/RestartDomoticz.sh"
    start program = "/etc/init.d/domoticz.sh start"
    stop  program = "/etc/init.d/domoticz.sh stop"
    if changed pid then exec "/usr/local/bin/monit2telegram"_

I would like to monitor zigbee2mqtt the same as i monitor domoticz.pid 
The problem is that i cannot find a pid like zigbee2mqtt.pid in the /var/run folder.

Is there a way to monitor zigbee2mqtt with Monit ?

**Comment by SargonofAssyria** [comment]
Something like, you have to experiment to get the right match
check process zigbee2mqtt matching "npm"
or
check program zigbee2mqtt with path "systemctl --quiet is-active zigbee2mqtt"
    if status != 0 then ...
You can find the pid by
sudo monit procmatch 'npm'

**Comment by lansi002** [comment]
Your solution works for me. Thank you for your help. 

     _check process zigbee2mqtt-service matching "npm"
     start program = "/usr/bin/sudo systemctl start zigbee2mqtt"
     stop  program = "/usr/bin/sudo systemctl stop zigbee2mqtt"
     if changed pid then exec "/usr/local/bin/monit2telegram"_

**Reply by abbychette** [reply]
Works fine for me. Thanks

**Comment by SargonofAssyria** [comment]
I had today an error that the adaptor could not be found anymore. No idea yet what caused it.
But Monit did not warn me, probably because the service is restarting to quickly.
Now I solve it by monitoring in Nodered if topic zigbee2mqtt/bridge/state has the state 'online'
